doc/ref/gappkg.xml

```diff
@@ -997,6 +997,57 @@ is now a needed package for &GAP; itself.
 </Section>
 
 
+<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% -->
+<Section Label="Extensions Provided by a Package">
+<Heading>Extensions Provided by a Package</Heading>
+
+Sometimes a package <C>A</C> can provide additional functionality,
+such as better methods or additional data,
+if some other packages <C>B</C>, <C>C</C>, etc. are loaded.
+However, one wants that package <C>A</C> can be used also without these
+packages,
+and therefore <C>B</C>, <C>C</C>, etc. shall not be regarded as needed
+packages (see Section <Ref Sect="Package dependencies"/>) of <C>A</C>.
+
+<P/>
+
+One way to deal with this situation is to put those parts of code of <C>A</C>
+that depend on <C>B</C>, <C>C</C>, etc., into files that get read only
+in the situation that the packages in question have actually been loaded
+into the current &GAP; session,
```

Comment:
Maybe stop the sentence here, and insert a brief excursion, something in this vein (exact text can of course be completely different)
```suggestion
into the current &GAP; session.
<P/>

However, this leaves the question when to load these files. In the past, the only
option for <C>A</C> was to check for the presence of <C>B</C>, <C>C</C>, etc., while
it itself was being loaded. The result of this is that which features are available or not depends on the order in which packages get loaded: If <C>B</C> is loaded before <C>A</C>, the extension might be loaded as well; if <C>B</C> is loaded only after <C>A</C>, then the extension is not loaded.
<P/>

To deal with this, &GAP; offers a dedicated mechanism for packages to deal
with extensions:
```